src/imap/mod.rs

```diff
@@ -768,17 +768,12 @@ impl Imap {
                 let msg = fetch?;
                 match get_fetch_headers(&msg) {
                     Ok(headers) => {
-                        let (from_id, _, _) = from_field_to_contact_id(
-                            context,
-                            &mimeparser::get_from(&headers),
-                            false,
-                        )
-                        .await?;
-                        if from_id == DC_CONTACT_ID_SELF {
```

Comment:
In this line, we checked that FROM is SELF before. At least it was supposed to, maybe this didn't work?